src/api-v1.js

```diff
@@ -70,100 +70,106 @@ api.declare({
     // Intentionally vague because we don't want to reveafollowRedirects
     // our configuration too much
     let msg;
+    let code = 503; // default is that something is temporarily wrong
     switch (err.code) {
-      case 'HTTPError':
-        msg = 'HTTP Error while trying to resolve redirects';
-        break;
       case 'DoesNotMatchPatterns':
         msg = 'Input URL does not match whitelist';
+        code = 403; // forbidden
         break;
       case 'InsecureURL':
         msg = 'Refusing to follow a non-SSL redirect';
+        code = 403; // forbidden
+        break;
+      case 'HTTPError':
+        msg = 'HTTP Error while trying to resolve redirects';
         break;
       default:
         msg = 'Input URL failed validation for unknown reason';
         break;
     }
-    return res.status(400).json({
+    return res.status(code).json({
       msg: msg,
+      code: code,
     });
   }
 
-  if (service.toLowerCase() === 's3') {
-    if (this.s3backends[region]) {
-      let backend = this.s3backends[region];
-
-      let maxWaitForCachedCopy = this.maxWaitForCachedCopy;
-
-      let startTime = new Date();
-      let x = 0;
-      while (new Date() - startTime < maxWaitForCachedCopy) {
-        debug(`Check ${x++} of ${url}`);
+  // This is the ID that we need to find a backend for
+  let incomingId = `${service}_${region}`;
```

Comment:
I wanted to avoid using characters that have a special meanting in a URL context, in case we wanted to make the internal ID an external concept.